src/Functional.CQS.AOP.IoC.PureDI.MetricsCapturing/Extensions/AsyncCommandHandlerExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Functional.CQS.AOP.IoC.PureDI.MetricsCapturing.Extensions
+{
+	internal static class AsyncCommandHandlerExtensions
```

Comment:
It looks like you're indirectly testing these extension methods via the MetricsCapturingDecorators. Is there any value to be gained from also testing these extension methods directly? For example, are there any assertions around the TimeElapsed that make sense?

Reply:
These extension methods are just a way to reduce code duplication between all the decorators.  The decorators already delegate the majority of the logic to the extension method, so I'm considering them sufficiently tested.